jpo-ode-svcs/src/main/java/us/dot/its/jpo/ode/udp/UdpHexDecoder.java

```diff
@@ -30,47 +23,49 @@
 import us.dot.its.jpo.ode.model.OdeTimMetadata;
 import us.dot.its.jpo.ode.model.ReceivedMessageDetails;
 import us.dot.its.jpo.ode.model.RxSource;
+import us.dot.its.jpo.ode.uper.SupportedMessageType;
 import us.dot.its.jpo.ode.uper.UperUtil;
+import us.dot.its.jpo.ode.util.DateTimeUtils;
 import us.dot.its.jpo.ode.util.JsonUtils;
 
+import java.net.DatagramPacket;
+
+@Slf4j
 public class UdpHexDecoder {
-    
-    private static Logger logger = LoggerFactory.getLogger(UdpHexDecoder.class);
-
-    public static OdeAsn1Payload getPayloadHexString(DatagramPacket packet, UperUtil.SupportedMessageTypes msgType) {
-      String startFlag = UperUtil.getStartFlag(msgType);
-      // extract the actual packet from the buffer
-      byte[] payload = packet.getData();
-      if (payload == null)
-         return null;
-      // convert bytes to hex string and verify identity
-      String payloadHexString = HexUtils.toHexString(payload).toLowerCase();
-      if (payloadHexString.indexOf(startFlag) == -1)
-         return null;
-      
-      logger.debug("Full {} packet: {}", msgType, payloadHexString);
-      
-      payloadHexString = UperUtil.stripTrailingZeros(UperUtil.stripDot3Header(payloadHexString, startFlag)).toLowerCase();
-      logger.debug("Stripped {} packet: {}", msgType, payloadHexString);
-
-      OdeAsn1Payload odePayload = new OdeAsn1Payload(HexUtils.fromHexString(payloadHexString));
-      
-      return odePayload;
+
+    private UdpHexDecoder() {
+        throw new UnsupportedOperationException();
     }
 
-    public static String buildJsonMapFromPacket(DatagramPacket packet){
+    public static OdeAsn1Payload getPayloadHexString(DatagramPacket packet, SupportedMessageType msgType) throws InvalidPayloadException {
+        // extract the actual packet from the buffer
+        byte[] payload = packet.getData();
+        if (payload == null)
+            throw new InvalidPayloadException("Payload is null");
+        // convert bytes to hex string and verify identity
+        String payloadHexString = HexUtils.toHexString(payload).toLowerCase();
+        if (!payloadHexString.contains(msgType.getStartFlag()))
+            throw new InvalidPayloadException("Payload does not contain start flag");
```

Comment:
Note: instead of returning `null`, we should return an exception here to align with [Clean Code's Error Handling tenants](https://medium.com/@himanshuganglani/clean-code-error-handling-d8562573d324). By throwing an exception instead of returning null, we can avoid NPEs and more clearly indicate when this method cannot get a proper hex string from the payload.